doc/content.tex

```diff
@@ -1,1349 +0,0 @@
-    predicted points.
-
-    \begin{lstlisting}[caption={Interface for time series models.}, language=Java, label={alg:models}]
-interface TimeSeriesModel {
-    void learn(List<DataPoint> ts);
-    List<DataPoint> forecast(int steps);
-    int numberOfParameters();
-    MetricContext context();
-    AccuracyStatistics initStatistics();
-    AccuracyStatistics runStatistics();
-    ...
-     }
-    \end{lstlisting}
-
-    List of implemented models, statistical and utility classes for time series manipulation:
-
-    \begin{multicols}{2}
-        \begin{itemize}
-            \item Simple ex. smoothing
-            \item Double ex. smoothing
-            \item Triple ex. smoothing
-            \item Weighted moving average
-            \item Automatic forecaster
-            \item Augmented Dickey\,--\,Fuller test
-            \item Time series decomposition
-            \item Autocorrelation function
-            \item Time series lagging
-            \item Time series differencing
-            \item Automatic period identification
-        \end{itemize}
-    \end{multicols}
-
-        %%%%%%%%
-        \subsection{Automatic Forecaster}
-        One of the most important forecasting classes in Data Mining is \\\texttt{AutomaticForecaster}. This class
-        autonomously decides which time series model should be used for modelled time series. Currently it decides from
-        three implemented models: simple, double and triple exponential smoothing. Other models which implement
-        interface \\\texttt{TimeSeriesModel} can be easy added.
-
-        The most importantly this class is capable of dealing with concept drift. It holds circular buffer of historical
-        metrics and if statistical properties of the underlying time series changes content of the buffer is used for
-        selecting new model.
-
-        The strategy when a new model should be selected is configurable. System currently
-        supports two strategies. The first strategy periodically each $n$ observations selects new model.
-        The second one is more sophisticated and it selects a new model only when the error produced on learning data
-        exceeds by $x\%$ error calculated on learning points when model was initlized.
-
-        Model selection is based on information criterion from section \ref{sec:model-quality}. Automatic forecaster in
-        successive steps calls optimizers of given models and then selects the best with the lowest information
-        criterion. Used criterion is configurable for each forecaster.
-
-        %%%%%%%%
-        \subsection{Model Optimizers}
-        Model optimizers are the most important part of the models implementation. If the model parameters are not
-        estimated correctly model produces high forecasting error.
-
-        The idea behind optimization is to find such
-        parameters best describes training data \cite{hyndman-state-space}. Optimization criterion can be: mean
```

Comment:
PREPAC, samozrejme describe, tu som narobila paseku v celej vete, ma byt:

..paremeters that describe training data the best.